thespian/system/transport/TCPTransport.py

```diff
@@ -340,13 +340,12 @@ def getAdminAddr(capabilities):
     @staticmethod
     def getAddressFromString(addrspec, adminRouting=False):
         if isinstance(addrspec, tuple):
-            addrparts = addrspec
+            addrparts = list(addrspec)
         else:
             addrparts = addrspec.split(':')
         addrtype = RoutedTCPv4ActorAddress if adminRouting else TCPv4ActorAddress
-        if 1 == len(addrparts):
-            return ActorAddress(addrtype(addrparts[0], DEFAULT_ADMIN_PORT, external=True))
-        return ActorAddress(addrtype(addrparts[0], addrparts[1], external=True))
+        ip, port = addrparts[0], addrparts[1] if len(attrparts) > 1 else DEFAULT_ADMIN_PORT
+        return ActorAddress(addrtype(ip, port, external=True))
```

Comment:
Nice!  You could even avoid the update of addrparts in inline the added default and evaluation.  

If you update the pull request (remove the list() on the addrspec tuple above, too) I'd like to get this merged.

P.S.  True the variable allocations are definitely minor, and akin to nothing next to time wasters like ```select()```, but I like keeping things tight... just like with your ```any()``` optimization below.